packages/orchestration/src/exos/cosmos-orchestration-account.js

```diff
@@ -259,6 +266,22 @@ export const prepareCosmosOrchestrationAccountKit = (
           return harden(toDenomAmount(balance));
         },
       },
+      allBalancesQueryWatcher: {
+        /**
+         * @param {JsonSafe<ResponseQuery>[]} results
+         */
+        onFulfilled([result]) {
+          // an empty string is a valid result
+          if ((result?.key ?? null) === null) {
+            throw Fail`Error parsing QueryAllBalances result ${q(result)}`;
+          }
+          const { balances } = QueryAllBalancesResponse.decode(
+            decodeBase64(result.key),
+          );
```

Comment:
important comment!

should we handle more failure more generally?
```suggestion
let response;
try {
  response = QueryAllBalancesResponse.decode(
    decodeBase64(result.key),
  );
} catch {
   throw Fail`Error parsing QueryAllBalances result ${q(result)}`;
}
const { balances } = response;
```

Reply:
Yes - was surprised by this! And good suggestion